misc/docker-ci/check.mk

```diff
@@ -3,7 +3,7 @@ SRC_DIR=/h2o
 CHECK_MK=$(SRC_DIR)/misc/docker-ci/check.mk
 CMAKE_ARGS=
 FUZZ_ASAN=ASAN_OPTIONS=detect_leaks=0
-DOCKER_RUN_OPTS=--privileged -v `pwd`:$(SRC_DIR) -v /sys/kernel/debug:/sys/kernel/debug -e TRAVIS --add-host=127.0.0.1.xip.io:127.0.0.1 -it
+DOCKER_RUN_OPTS=--privileged -v `pwd`:$(SRC_DIR) -v /sys/kernel/debug:/sys/kernel/debug --add-host=127.0.0.1.xip.io:127.0.0.1
```

Comment:
I think the only concerns regarding this pull request are in the changes to this file.
* Re removing `-it`, I agree that we do not need to open TTY, and I think it's correct to drop them. If anything bad happens, we can revive them.
* Re removing `-e TRAVIS`, discussed offline with @gfx.

Reply:
Re `-it`: It seems that GitHub Actions have different configurations from Travis about TTY, so `docker run -it` fails with the error "the input device is not a TTY", so we should find workarounds if it gets required.

Reply:
Thank you for the clarification. If some environments happen to require the use of `-it`, we can always pass that as an argument to make running this makefile.